cartographer_rviz/cartographer_rviz/submaps_display.cc

```diff
@@ -172,10 +177,18 @@ void SubmapsDisplay::processMessage(
     }
     trajectory_submaps.at(id.submap_index)->Update(msg->header, submap_entry);
   }
+  // Remove all deleted trajectories not mentioned in the SubmapList.
+  for (auto it = trajectories_.begin(); it != trajectories_.end();) {
+    if (listed_trajectories.count(it->first) == 0) {
+      trajectories_.erase(it++);
+    } else {
+      ++it;
+    }
+  }
   // Remove all submaps not mentioned in the SubmapList.
-  for (size_t trajectory_id = 0; trajectory_id < trajectories_.size();
-       ++trajectory_id) {
-    auto& trajectory_submaps = trajectories_[trajectory_id]->submaps;
+  for (const auto& trajectory_by_id : trajectories_) {
```

Comment:
if you have a better name for this `std::pair<int, cartographer_rviz::Trajectory>` than "trajectory_by_id` - let me know